File: Domain/Models/Inventory/Product.cs
using System.ComponentModel.DataAnnotations;

namespace Domain.Models.Inventory
{
    public class Product
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required, StringLength(50)]
        public string Sku { get; set; } = string.Empty;

        [StringLength(50)]
        public string? Barcode { get; set; }

        [Required, StringLength(250)]
        public string NameAr { get; set; } = string.Empty;

        [StringLength(250)]
        public string? NameEn { get; set; }

        [StringLength(1000)]
        public string? Description { get; set; }

        public Guid CategoryId { get; set; }
        public Category? Category { get; set; }

        public Guid UnitId { get; set; }
        public Unit? Unit { get; set; }

        public decimal PurchasePrice { get; set; }
        public decimal SalePrice { get; set; }
        public decimal MinSalePrice { get; set; }

        // Egyptian VAT - default 14%
        public decimal VatRate { get; set; } = 14m;

        // ETA classification codes for Egyptian e-invoicing
        [StringLength(50)]
        public string? ItemCode { get; set; }

        [StringLength(50)]
        public string? GS1Code { get; set; }

        public decimal MinStockLevel { get; set; }
        public decimal MaxStockLevel { get; set; }

        public bool TrackStock { get; set; } = true;
        public bool IsActive { get; set; } = true;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }

        public ICollection<StockItem>? StockItems { get; set; }
    }
}

File: Mobile/lib/features/inventory/models/product.dart
class Product {
  final String id;
  final String sku;
  final String? barcode;
  final String nameAr;
  final String? nameEn;
  final String? description;
  final String categoryId;
  final String? categoryName;
  final String unitId;
  final String? unitName;
  final double purchasePrice;
  final double salePrice;
  final double minSalePrice;
  final double vatRate;
  final double currentStock;
  final bool trackStock;
  final bool isActive;

  Product({
    required this.id,
    required this.sku,
    this.barcode,
    required this.nameAr,
    this.nameEn,
    this.description,
    required this.categoryId,
    this.categoryName,
    required this.unitId,
    this.unitName,
    required this.purchasePrice,
    required this.salePrice,
    required this.minSalePrice,
    required this.vatRate,
    required this.currentStock,
    required this.trackStock,
    required this.isActive,
  });

  factory Product.fromJson(Map<String, dynamic> json) => Product(
        id: json['id'],
        sku: json['sku'] ?? '',
        barcode: json['barcode'],
        nameAr: json['nameAr'] ?? '',
        nameEn: json['nameEn'],
        description: json['description'],
        categoryId: json['categoryId'],
        categoryName: json['categoryName'],
        unitId: json['unitId'],
        unitName: json['unitName'],
        purchasePrice: (json['purchasePrice'] ?? 0).toDouble(),
        salePrice: (json['salePrice'] ?? 0).toDouble(),
        minSalePrice: (json['minSalePrice'] ?? 0).toDouble(),
        vatRate: (json['vatRate'] ?? 14).toDouble(),
        currentStock: (json['currentStock'] ?? 0).toDouble(),
        trackStock: json['trackStock'] ?? true,
        isActive: json['isActive'] ?? true,
      );
}
